src/server/api/routers/automation.ts

```diff
@@ -0,0 +1,80 @@
+import { z } from "zod";
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+
+const createAutomationSchema = z.object({
+  name: z.string(),
+  desc: z.string(),
+  webhookID: z.string(),
+  condition: z.enum(["issues", "pull_request"]),
+  actionType: z.enum(["email"])
+})
+
+export const automationRouter = createTRPCRouter({
+  createAutomation: protectedProcedure.input(createAutomationSchema).mutation(async ({ ctx, input }) => {
+    const userID = ctx.session.user.id;
+  
+    const action = await ctx.prisma.action.create({
+      data: {
+        hookID: input.webhookID,
+        actionType: input.actionType,
+        toEmail: "some email", // input.toEmail,
+        subject: "some subject", // input.subject,
+        content: "some content" // input.content,
+      },
+    });
+  
+    const automation = await ctx.prisma.automation.create({
+      data: {
+        webhookID: input.webhookID,
+        name: input.name,
+        desc: input.desc,
+        status: "active",
+        condition: input.condition,
+        action: { connect: { id: action.id } },
```

Comment:
so users are limited to 1 action per automation?

Reply:
A user can have up to 10 WebHooks in a single repository but if any overlap, then you will get a 422 (Hook already exists in this repo). For example, you can have a WebHook that listens for PR's and another that listens for Issues and Stars in the same repo but you cannot have a WebHook that listens for PR's and another that listens for PR's and Issues.

Reply:
Oh, i meant about that line specifically.

```
action: { connect: { id: action.id } },
```

Does this mean for each row for automation in the DB there is exactly 1 corresponding action? Do you intend to handle cases whens someone says "when X happens do both Y and Z", or are users going to have to make 2 automations "when X happens do Y" and "when X happens do Z"

Reply:
I see. As of right now, I'm not accounting for this but this can be possible. As of right now, the latter would work but if we decide that the former is valuable to have, then we can alter the adjustments of what should be prioritized.